zoo/checklists/views.py

```diff
@@ -0,0 +1,127 @@
+import attr
+from django import forms
+from django.core.exceptions import SuspiciousOperation
+from django.http import Http404, JsonResponse
+from django.views.generic import ListView, TemplateView
+
+from . import models
+from ..services import models as service_models
+from .steps import STEPS
+
+
+def get_service_steps(service):
+    return [(tag, steps) for tag, steps in STEPS.items() if tag in service.tags]
+
+
+class GlobalChecklistsView(ListView):
+    model = service_models.Service
+    template_name = "checklists/global_checklists.html"
+
+    def get_queryset(self):
+        return (
+            super()
+            .get_queryset()
+            .filter(status=service_models.Status.BETA.value)
+            .prefetch_related("checkmarks")
+        )
+
+    def get_context_data(self, *, object_list=None, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["service_list"] = [
+            {
+                "name": service.name,
+                "owner_slug": service.owner_slug,
+                "name_slug": service.name_slug,
+                "completed_checklist_steps": service.checkmarks.count,
+                "total_checklist_steps": sum(
+                    [len(steps) for tag, steps in get_service_steps(service)]
+                ),
+            }
+            for service in self.get_queryset()
+        ]
+        return context
+
+
+class ServiceChecklistView(TemplateView):
+    template_name = "checklists/service_checklist.html"
+
+    def get_context_data(self, **kwargs):
+        service_owner_slug = kwargs.get("service_owner_slug")
+        service_name_slug = kwargs.get("service_name_slug")
+
+        if None in [service_owner_slug, service_name_slug]:
```

Comment:
Wat? `if not service_owner_slug or not service_name_slug` is a much more logical way of writing this. Or am I misunderstanding the condition?